src/berry_pots_app_tasks.c

```diff
@@ -0,0 +1,444 @@
+#include "global.h"
+#include "berry_pots_app.h"
+#include "berry_pots_app_internal.h"
+#include "unk_02005D10.h"
+#include "touchscreen.h"
+#include "system.h"
+#include "constants/sndseq.h"
+
+typedef struct SysTaskArgs_ov17_0220351C {
+    BerryPotsAppData *data;
+    u8 unk4;
+    u8 unk5;
+    u8 unk6;
+    u8 unk7;
+    u8 unk8;
+    s8 sign;
+    u8 unkA;
+} SysTaskArgs_ov17_0220351C;
+
+static const TouchscreenHitbox ov17_02203FC0 = {
+    .rect = {160, 192, 192, 0}
+};
+
+static void ov17_022035A4(SysTask *task, SysTaskArgs_ov17_0220351C *args);
+
+void ov17_0220351C(BerryPotsAppData *data) {
+    SysTaskArgs_ov17_0220351C *args = AllocFromHeap(data->heapId, sizeof(SysTaskArgs_ov17_0220351C));
+    MI_CpuFill8(args, 0, sizeof(SysTaskArgs_ov17_0220351C));
+    args->data = data;
+    args->unk6 = data->unk80;
+    args->unk7 = data->unk81;
+    s8 diff = data->unk81 - data->unk80;
+    if (diff > 0) {
+        args->sign = 1;
+    } else {
+        args->sign = -1;
+        diff = -diff;
+    }
+
+    args->unk8 = (s8)(diff * 27) / 4;
+    args->unkA = data->unk80;
+    SysTask_CreateOnMainQueue((SysTaskFunc)ov17_022035A4, args, 0);
+    data->runningTasks++;
+}
+
+static void ov17_022035A4(SysTask *task, SysTaskArgs_ov17_0220351C *args) {
+    BerryPotsAppData *unk0 = args->data;
+
+    Sprite_AddPositionXY(unk0->sprites[2], args->sign * 4, 0);
+    s16 x, y;
+    Sprite_GetPositionXY(unk0->sprites[2], &x, &y);
+
+    u8 potIndex = (x - 31) / 27;
+    if (potIndex != args->unkA && potIndex != args->unk7) {
+        ov17_02202B58(unk0, potIndex);
+        args->unkA = potIndex;
```

Comment:
can unkA be documented?

Reply:
not yet, would need substantial documentation of this code first